# 1.2 46

In [73]:
import numpy as np  

V= np.array([0, 2, 4, 6, 8, 10])  

F= np.array([0, 2.90, 14.8, 39.6, 74.3, 119])  

P5 = np.polyfit(V, F, 5)  

print(P5)  

t = 7.5  

f_1 = np.polyval(P5, t)  

print(f_1)  

F_1= f_1 * 100  

print(F_1)  


[ 2.60416667e-03 -7.00520833e-02  6.61458333e-01 -1.19479167e+00
  1.71250000e+00  2.08856057e-13]
64.83837890625014
6483.837890625015


The interpolation polynomial is:

$$
P(t)=0+1.7125t-1.1948t^2+0.6615t^3-0.0701t^4+0.0026t^5
$$

P(7.5)$\approx$6484(1b)

In [72]:
import numpy as np  

V= np.array([0, 2, 4, 6, 8, 10])  

F= np.array([0, 2.90, 14.8, 39.6, 74.3, 119])  

P3 = np.polyfit(V, F, 3)  

print(P3)  

f_3 = np.polyval(P3, t)  

print(f_3)  

F_3= f_3 * 100  

print(F_3)  


[-0.00462963  1.40962302 -1.7412037   0.17539683]
64.45453869047624
6445.453869047624


A cubic least-squares fit gives approximately:

$$
P3(t)=0.1754-1.7412t+1.4096t^2-0.0046t^3
$$

P3(7.5)$\approx$6445(1b)

A cubic polynomial cannot interpolate all six points simultaneously.

# 1.7 47-50

## 47

In [19]:
import sympy as sp
A_47 = sp.Matrix([
    [8, -3, 0, -7, 2],   
    [-9, 4, 5, 11, -7],   
    [6, -2, 2, -4, 4],    
    [5, -1, 7, 0, 10]   
]) 
rref_47= A_47.rref()
print(rref_47)

(Matrix([
[1, 0, 3, 1, 0],
[0, 1, 8, 5, 0],
[0, 0, 0, 0, 1],
[0, 0, 0, 0, 0]]), (0, 1, 4))


Therefore,

In [25]:
B_47 = A_47[:, [0,1,4]]
print(B_47)

Matrix([[8, -3, 2], [-9, 4, -7], [6, -2, 4], [5, -1, 10]])


In [26]:
nullspace_B_47 = B_47.nullspace()
print(nullspace_B_47)

[]


Therefore,The system has only the trivial solution.

## 48

In [43]:

A_48 = sp.Matrix([
    [12, 10, -6, -3, 7, 10],  
    [-7, -6, 4, 7, -9, 5],     
    [9, 9, -9, -5, 5, -1],     
    [-4, -3, 1, 6, -8, 9],     
    [8, 7, -5, -9, 11, -8]    
]) 

rref_48= A_48.rref()
print(rref_48)

(Matrix([
[1, 0,  2, 0,  2, 0],
[0, 1, -3, 0, -2, 0],
[0, 0,  0, 1, -1, 0],
[0, 0,  0, 0,  0, 1],
[0, 0,  0, 0,  0, 0]]), (0, 1, 3, 5))


Therefore,

In [30]:
B_48 = A_48[:, [0,1,3,5]]
print(B_48)

Matrix([[12, 10, -3, 10], [-7, -6, 7, 5], [9, 9, -5, -1], [-4, -3, 6, 9], [8, 7, -9, -8]])


In [31]:
nullspace_B_48 = B_48.nullspace()
print(nullspace_B_48)

[]


Therefore,The system has only the trivial solution.

## 49

Choose:
v=[0,5,2,7]

Solve:BX=V

In [81]:
import sympy as sp
B=sp.Matrix([[8, -3, 2], 
            [-9, 4, -7], 
            [6, -2, 4], 
            [5, -1, 10]])
v=sp.Matrix([0,5,2,7])
ans=sp.linsolve((B,v))
print(ans)

{(3, 8, 0)}


The solution is:
x=[3,8,0]

Therefore:
v=a3=3b1+8b2

Hence:

v is in the set spanned by the columns of B

## 50

Choose:
v=[-6,4,-9,1,-5]

Solve:BX=V

In [82]:
import sympy as sp
B=sp.Matrix([[12, 10, -3, 10], 
             [-7, -6, 7, 5], 
             [9, 9, -5, -1], 
             [-4, -3, 6, 9], 
             [8, 7, -9, -8]])
v=sp.Matrix([-6,4,-9,1,-5])
ans=sp.linsolve((B,v))
print(ans)

{(2, -3, 0, 0)}


The solution is:X=[2,-3,0,0]

Therefore:v=a3=2b1-3b2

Hence:

v is in the set spanned by the columns of B

# 1.8 38

A linear transformation must satisfy:T(0)=0

Now evaluate the T(0):
T(0)=A*0+b=b

When:
$b\neq0$

Therefore:
$
T(0)=b\neq0
$

Therefore:
T(X)=Ax+b is not linear when $b\neq0$

# 1.9 45-48

In [70]:
import sympy as sp

A45 = sp.Matrix([
    [-5,10,-5,4],
    [8,3,-4,7],
    [4,-9,5,-3],
    [-3,-2,5,4]
])

A46 = sp.Matrix([
    [7,5,4,-9],
    [10,6,16,-4],
    [12,8,12,7],
    [-8,-6,-2,5]
])

A47 = sp.Matrix([
    [4,-7,3,7,5],
    [6,-8,5,12,-8],
    [-7,10,-8,-9,14],
    [3,-5,4,2,-6],
    [-5,6,-6,-7,3]
])

A48 = sp.Matrix([
    [9,13,5,6,-1],
    [14,15,-7,-6,4],
    [-8,-9,12,-5,-9],
    [-5,-6,-8,9,8],
    [13,14,15,2,11]
])

A45_rref=A45.rref()[0]
A46_rref=A46.rref()[0]
A47_rref=A47.rref()[0]
A48_rref=A48.rref()[0]

print(A45_rref)
print(A46_rref)
print(A47_rref)
print(A48_rref)

Matrix([[1, 0, 0, 44/35], [0, 1, 0, 79/35], [0, 0, 1, 86/35], [0, 0, 0, 0]])
Matrix([[1, 0, 7, 0], [0, 1, -9, 0], [0, 0, 0, 1], [0, 0, 0, 0]])
Matrix([[1, 0, 0, 5, 0], [0, 1, 0, 1, 0], [0, 0, 1, -2, 0], [0, 0, 0, 0, 1], [0, 0, 0, 0, 0]])
Matrix([[1, 0, 0, 0, 5], [0, 1, 0, 0, -4], [0, 0, 1, 0, 0], [0, 0, 0, 1, 1], [0, 0, 0, 0, 0]])


For A45_rref:
There are 4 columns, but only 3 pivots.

Therefore:A45 is not one-to-one

For A46_rref:
There are 4 columns, but only 3 pivots.

Therefore:A46 is not one-to-one

For A47_rref:
There are 5 rows, but only 4 pivots.

Therefore:A47 is not onto $R^5$

For A48_rref:
There are 5 rows, but only 4 pivots.

Therefore:A48 is not onto $R^5$

# 1.10 12

In [85]:
import sympy as sp  

M = sp.Matrix([[0.97, 0.05, 0.10], 
              [0.00, 0.90, 0.05],  
              [0.03, 0.05, 0.85]])

x1 = sp.Matrix([295, 55, 150])  

x2 = M*x1  

x3 = M*x2  

print(x2) 

print(x3)  

Matrix([[303.900000000000], [57.0000000000000], [139.100000000000]])
Matrix([[311.543000000000], [58.2550000000000], [130.202000000000]])


Therefore, Wednesday's approximate distribution is:


Airport:312 cars


East:58 cars


West:130 cars

# 1.10 13

## a

In [86]:
import sympy as sp

M=sp.Matrix([[0.95,0.03],
           [0.05,0.97]])

x=sp.Matrix([600000,400000])

for i in range(1,21):
    x= M * x
    print(i,x)
    

1 Matrix([[582000.000000000], [418000.000000000]])
2 Matrix([[565440.000000000], [434560.000000000]])
3 Matrix([[550204.800000000], [449795.200000000]])
4 Matrix([[536188.416000000], [463811.584000000]])
5 Matrix([[523293.342720000], [476706.657280000]])
6 Matrix([[511429.875302400], [488570.124697600]])
7 Matrix([[500515.485278208], [499484.514721792]])
8 Matrix([[490474.246455951], [509525.753544049]])
9 Matrix([[481236.306739475], [518763.693260525]])
10 Matrix([[472737.402200317], [527262.597799683]])
11 Matrix([[464918.410024292], [535081.589975708]])
12 Matrix([[457724.937222348], [542275.062777651]])
13 Matrix([[451106.942244561], [548893.057755439]])
14 Matrix([[445018.386864996], [554981.613135004]])
15 Matrix([[439416.915915796], [560583.084084204]])
16 Matrix([[434263.562642532], [565736.437357467]])
17 Matrix([[429522.477631130], [570477.522368870]])
18 Matrix([[425160.679420639], [574839.320579360]])
19 Matrix([[421147.825066988], [578852.174933012]])
20 Matrix([[417455.99

The results show:  
City population decreases over time.  
Suburban population increases over time.  
The system approaches a stable distribution.  

## b

In [88]:
import sympy as sp 

M = sp.Matrix([[0.95, 0.03],  
              [0.05, 0.97]])

x = sp.Matrix([350000, 650000])  

for i in range(1, 21):  
    x = M * x  
    
    print(i,x)  

1 Matrix([[352000.000000000], [648000.000000000]])
2 Matrix([[353840.000000000], [646160.000000000]])
3 Matrix([[355532.800000000], [644467.200000000]])
4 Matrix([[357090.176000000], [642909.824000000]])
5 Matrix([[358522.961920000], [641477.038080000]])
6 Matrix([[359841.124966400], [640158.875033600]])
7 Matrix([[361053.834969088], [638946.165030912]])
8 Matrix([[362169.528171561], [637830.471828439]])
9 Matrix([[363195.965917836], [636804.034082164]])
10 Matrix([[364140.288644409], [635859.711355590]])
11 Matrix([[365009.065552856], [634990.934447143]])
12 Matrix([[365808.340308628], [634191.659691372]])
13 Matrix([[366543.673083937], [633456.326916062]])
14 Matrix([[367220.179237222], [632779.820762777]])
15 Matrix([[367842.564898245], [632157.435101755]])
16 Matrix([[368415.159706385], [631584.840293614]])
17 Matrix([[368941.946929874], [631058.053070125]])
18 Matrix([[369426.591175484], [630573.408824515]])
19 Matrix([[369872.463881445], [630127.536118554]])
20 Matrix([[370282.66

The system also approaches a stable distribution. 

Now caculate the stable state:  

In [92]:
import sympy as sp 
M = sp.Matrix([[0.95, 0.03],  
              [0.05, 0.97]])

x = sp.Matrix([600000, 400000])  

for i in range(1, 1000):  
    x = M * x  
    
print(x)  

Matrix([[374999.999999994], [624999.999999991]])


In [93]:
import sympy as sp 
M = sp.Matrix([[0.95, 0.03],  
              [0.05, 0.97]])

x = sp.Matrix([350000, 650000])  

for i in range(1, 1000):  
    x = M * x  
    
print(x)  

Matrix([[374999.999999994], [624999.999999993]])


Therefore:  
The long-term distribution is:  
37.5% city and 62.5% suburbs